### PR TITLE
Documentation for the things I learned in #9

### PR DESCRIPTION
**Pull request by moonglum**
I learned some things in #9 from @frodsan – time to add it to the documentation:

1. How to contribute to the documentation. I added this info to the CONTRIBUTING file.
2. How to add middlewares to Tynn and how to support Method Overriding. I added this info to a new "middlewares" page.

Thanks for teaching me :heart:

### CODE REVIEW

File: docs/guides/middlewares.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+HTML Forms currently only support GET and POST requests. You may want to have a form that performs other actions such as PUT though. The usual solution to simulate a PUT is using a POST form, but adding a hidden input field with the name `_method` and the value `put`. For example:
+
+```html
+<form method="post" action="/update">
+  <input type="hidden" name="_method" value="put">
+  ...
+</form>
+```
+
+In Syro this would however trigger the `post` action, as it is send as a post request from your browser. With a middleware we can now rewrite this request and change the method to be `PUT`. There is a middleware called [Rack::MethodOverride](https://github.com/rack/rack/blob/master/lib/rack/method_override.rb) included in Rack that does exactly that, so let's add it to our Syro app:
```
Comment (frodsan) [inline docs/guides/middlewares.md]: s/Syro/Tynn? :-)

**Comment by frodsan** [comment]
Looks perfect! Thanks! :heart:

**Comment by moonglum** [comment]
Fixed. You're welcome :smile_cat:

**Comment by moonglum** [comment]
Yay :dancers:

**Pull request merged**